### PR TITLE
Adding max topic with MAX_TOPIC and topic_count as parameters

### PR DESCRIPTION
**Pull request by Xenonar**
Resubmit the first pull request about max topic and topic counter

### CODE REVIEW

**Comment by Xenonar** [comment]
I have push another commit. For garbage collection. I cannot make another pull request because it automatic update. I think.

**Comment by herjulf** [comment]
Thanks,
It is a hard to follow with the cleaning up patch.
I would like to see this PR to be very clean.
You still have the code handy? If so I can close this PR. And you resubmit a new clean PR. Is that OK?

**Comment by herjulf** [comment]
OK. Thanks. I'll merge this PR.
This is from CSD course 2018 IoT team. It wass supported by the the original pubsub authors from KTH. Ramcin and Pethrus.
Some evaluation is results are in the project report.
